test/integration/collection-management/collection.test.ts

```diff
@@ -95,18 +95,14 @@ describe('Collection', function () {
       ]);
     });
 
-    it('should fail due to illegal listCollections', function (done) {
-      expect(() => db.collection(5)).to.throw('Collection name must be a String');
-      expect(() => db.collection('')).to.throw('Collection names cannot be empty');
-      expect(() => db.collection('te$t')).to.throw("Collection names must not contain '$'");
-      expect(() => db.collection('.test')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test.')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test..t')).to.throw('Collection names cannot be empty');
-      done();
+    it('should fail on server due to illegal collection name', async function () {
+      try {
+        const illegalCollection = await db.createCollection('a\x00b');
```

Comment:
Because we're using createCollection here which actually reaches out to the server to create the collection we aren't going to reach the next line performing the insert. I think we want the `db.collection` helper here, right?

Reply:
I see, I didn't realize it fails prior, I'll change it to `db.collection `